netbox/extras/migrations/0108_script_models.py

```diff
@@ -0,0 +1,92 @@
+    ct_script_module = ContentType.objects.filter(app_label='extras', model='scriptmodule').first()
+    for rule in EventRule.objects.filter(action_object_type=ct_script_module):
+        name = rule.action_parameters.get('script_name')
+        obj, created = Script.objects.get_or_create(
+            module_id=rule.action_object_id,
+            name=name,
+            defaults={'is_valid': False}
+        )
+
+        rule.action_object_type = ct
+        rule.action_object_id = obj.id
+        rule.save()
+
+
+def update_scripts(apps, schema_editor):
+    from extras.models import ScriptModule
+    ScriptModuleNew = apps.get_model('extras', 'ScriptModule')
+    Script = apps.get_model('extras', 'Script')
+    ContentType = apps.get_model('contenttypes', 'ContentType')
+    ct = ContentType.objects.filter(app_label='extras', model='script').first()
+
+    for module in ScriptModule.objects.all():
+        for script in module.get_module_scripts.keys():
+            obj = Script.objects.create(
+                name=script,
+                module=ScriptModuleNew.objects.get(file_root=module.file_root, file_path=module.file_path),
+            )
+
+            # update all jobs associated with this module/name to point to the new script obj
+            if ct:
+                module.jobs.filter(name=script).update(object_type=ct, object_id=obj.id)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('extras', '0107_convert_reports_to_scripts'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Script',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False)),
+                ('name', models.CharField(max_length=79)),
+                ('module', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='scripts', to='extras.scriptmodule')),
+            ],
+            options={
+                'ordering': ('name', 'pk'),
+            },
+        ),
+        migrations.AddField(
+            model_name='script',
+            name='is_valid',
+            field=models.BooleanField(default=True),
+        ),
+        migrations.AddConstraint(
+            model_name='script',
+            constraint=models.UniqueConstraint(fields=('name', 'module'), name='extras_script_unique_name_module'),
+        ),
+        migrations.RunPython(
+            code=update_scripts,
+            reverse_code=migrations.RunPython.noop
+        ),
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunPython(
+                    code=update_event_rules,
+                    reverse_code=migrations.RunPython.noop
+                ),
+            ],
+            state_operations=[
+                migrations.RemoveField(
+                    model_name='eventrule',
+                    name='action_parameters',
+                ),
+            ]
```

Comment:
Why separate state here? With the current approach, it leaves the `action_parameters` on the table; don't we want to remove it?